components/BottomSheet/src/MDCBottomSheetPresentationController.h

```diff
@@ -0,0 +1,40 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ A UIPresentationController for presenting a modal view controller as a bottom sheet.
+ 
+ @note If your presented view controller has a UIScrollView as the root view (or you are presenting
+ a UICollectionViewController or UIWebViewController) MDCBottomSheetPresentationController will 
+ observe the contentSize of the scroll view in order to react to changes in the content size.
+ @note MDCBottomSheetPresentationController can dismiss the presented view controller if the user
+ taps outside the bottom sheet or pulls it offscreen.
+ */
+@interface MDCBottomSheetPresentationController : UIPresentationController
+
+/**
+ Controls the height that the sheet should be when it appears. If NO, it defaults to half the height
+ of the screen. If YES, and the @c -preferredContentSize is non-zero on the content view controller,
+ it defaults to the preferredContentSize.
+
+ @note The height used will never be any taller than the screen of the device.
+ The default value is NO.
+ */
+@property(nonatomic) BOOL usePreferredHeight;
```

Comment:
I think you can remove this property and make it always use the preferredContentSize of the VC. I only added this property in the GK implementation, because I had already committed to a "half the container view height" heuristic early on that some clients were relying on. In actual fact, most clients will want to explicitly control how much of the view is exposed by the sheet when it appears